src/auxiliary/p4est.jl

```diff
@@ -118,7 +118,14 @@ coarsen_p4est!(p8est::Ptr{p8est_t}, recursive, coarsen_fn_c, init_fn_c) = p8est_
 # 2D
 ghost_new_p4est(p4est::Ptr{p4est_t}) = p4est_ghost_new(p4est, P4est.P4EST_CONNECT_FACE)
 # 3D
-ghost_new_p4est(p8est::Ptr{p8est_t}) = p8est_ghost_new(p8est, P4est.P8EST_CONNECT_FACE)
+# We need `P8EST_CONNECT_FULL` here, otherwise the ghost layer does not always hold all
```

Comment:
Thanks a lot for the detailed comment and congrats on tracking down the bug in your code. Do you know whether there is another way to fix this (with reasonable amount of code) other than requiring a full ghost layer? Also, the referenced comment in p4est only reports the reason for the small remote cell being `-1` - do you know that using `P8EST_CONNECT_FULL` really fixes the issue or did you just test it and it seems to fix it? In the latter case, I suggest to create an issue at the p4est repo and ask if your solution is indeed a true fix and not just working by chance.

Reply:
I just tested it and it seemed to fix the issue. But I did investigate it a little bit more now and my understanding of the issue is as follows. Consider the neighbor elements of a mortar in 3D.
```
 ┌─────────────┬─────────────┐  ┌───────────────────────────┐
 │             │             │  │                           │
 │    small    │    small    │  │                           │
 │      3      │      4      │  │                           │
 │             │             │  │           large           │
 ├─────────────┼─────────────┤  │             5             │
 │             │             │  │                           │
 │    small    │    small    │  │                           │
 │      1      │      2      │  │                           │
 │             │             │  │                           │
 └─────────────┴─────────────┘  └───────────────────────────┘
```
Suppose one process only owns element 4 (this is actually what happened in my case). Now the problem is that element 1 is not connected via a face to element 4, hence it won't be in the ghost layer if it is constructed with `P8EST_CONNECT_FACE`.
So yes, I think this should really fix the issue. I'll update the comment with this explanation.

Reply:
Great explanation, thanks! And yes, please put it in the comment!

Reply:
Done in [8bc24d8](https://github.com/trixi-framework/Trixi.jl/pull/1091/commits/8bc24d8526cb87d2cc4d4229f28c5493da517af4)

Reply:
Sounds already good! Could you please add also the link to the p4est code? I think that might come in handy in the future.